src/main/java/de/tum/in/www1/artemis/domain/lti/Lti13DeepLinkingResponse.java

```diff
@@ -54,19 +64,28 @@ public class Lti13DeepLinkingResponse {
 
     /**
      * Constructs an Lti13DeepLinkingResponse from an OIDC ID token and client registration ID.
+     * The 'aud' and 'iss' fields are reversed in the response compared to the request. This is a specific requirement in LTI 1.3 Deep Linking:
+     * - The 'iss' (Issuer) claim in the Deep Linking Request becomes the 'aud' (Audience) claim in the Response.
+     * - The 'aud' claim in the Request becomes the 'iss' claim in the Response.
+     * This reversal ensures that the response is directed to and validated by the correct entity, adhering to the OIDC and LTI 1.3 protocols.
      *
      * @param ltiIdToken           the OIDC ID token
      * @param clientRegistrationId the client registration ID
      */
     public Lti13DeepLinkingResponse(OidcIdToken ltiIdToken, String clientRegistrationId) {
         validateClaims(ltiIdToken);
 
-        this.deepLinkingSettings = JsonParser.parseString(ltiIdToken.getClaim(Claims.DEEP_LINKING_SETTINGS).toString()).getAsJsonObject();
+        Map<String, Object> deepLinkingSettings = ltiIdToken.getClaim(Claims.DEEP_LINKING_SETTINGS);
+        // convert the map to json
+        this.deepLinkingSettings = new GsonBuilder().setPrettyPrinting().create().toJsonTree(deepLinkingSettings).getAsJsonObject();
```

Comment:
Do we need pretty printing here? Is this JSON at some place visible for users? How is it used in general?

Reply:
Thank you for noticing! We do not necessarily need pretty printing here. It is now fixed.